File: Leetcode/26-Remove-Duplicates-from-Sorted-Array/RemoveDuplicatesFromSortedArray.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        if (nums.size() <1) {
            return i;
        }
        int j = i+1;
        int cur = nums.at(i);
        int size = nums.size();
        while (j<size) {
            if (nums.at(j) != cur) {
                i++;
                nums.at(i) = nums.at(j);
                cur = nums.at(i);
            }
            j++;
        }
        return i+1;
    }
};

File: Leetcode/26-Remove-Duplicates-from-Sorted-Array/RemoveDuplicatesFromSortedArray.go
func removeDuplicates(nums []int) int {
	i := 0
	if nums == nil || len(nums) < 1 {
		return i
	}

	j := i + 1
	cur := nums[i]
	size := len(nums)
	for j < size {
		if nums[j] != cur {
			i++
			nums[i] = nums[j]
			cur = nums[i]
		}
		j++
	}
	return i + 1
}
